### ISSUE TITLE
Centralized Roadmap

### ISSUE THREAD

**Issue by marcuscastelo** [marcuscastelo/marucs-diet#488]
Centralized Roadmap

This roadmap centralizes all improvements, features, and initiatives of the project organized by version and category to facilitate prioritization and planning. It includes all open issues of the project (limit 1000) organized strategically.

---

### 🚀 v0.12.0 - Major Stability and UX Improvements
**Status**: Completed milestone with mobile app foundation
**Focus**: Critical bug fixes, architecture refactoring, performance optimizations, and Capacitor-based mobile app MVP

- [x] Major architecture migrations and performance improvements
- [x] Critical bug fixes across multiple modules  
- [x] Mobile app MVP with Capacitor integration
- [x] UI/UX improvements and standardization
- [x] Code quality and maintenance improvements

**Remaining Open Issues**:
- [x] **#743** Recipe search add ignores user quantity, always uses default (preparedMultiplier not respected) - *bug/medium complexity/needs-investigation*
- [x] **#891** Loading States and Visual Feedback for Profile Charts - *complexity-low/task/performance/ui/mobile*
- [x] **#890** Shared Data Context for Profile Charts Performance - *feature/complexity-high/performance/backend/mobile*
- [x] **#889** Progressive Chart Loading for Profile Page Performance - *complexity-medium/task/performance/ui/mobile*
- [x] **#870** Test APK on real Android devices - *complexity-low/task/mobile*

---

### 🎯 v0.13.0 - Critical Domain Refactoring and Essential Features
**Status**: Active Development  
**Focus**: Foundational improvements, legacy code cleanup, and core functionality stabilization

#### 🏗️ Architecture & Domain Refactoring
- [x] **#730** Reduce schema duplication across all domain modules - *refactor*
- [x] **#727** Unify and synchronize macro/weight logic across DayMacros, macroOverflow, MaxQuantityButton, WeightEvolution, and related components - *refactor*
- [x] **#733** Calculate macros based on the quantity and reference - *refactor/domain*
- [x] **#723** Remove dayId from functions that don't need it - *refactor/application*
- [x] **#713** Implement proper infrastructure folder for recent food - *application/infra*

#### 🧹 Domain Code Cleanup (Legacy Utility Removal)
- [x] **#672** Issue: Remove Legacy Utility Usage from Domain/Application Code - *refactor/domain/application/code-cleanup*
- [x] **#707** Refactor: Remove deepCopy Usage from Domain/Application Code - *refactor/domain/application*
- [ ] **#706** Refactor: Remove macroMath Usage from Domain/Application Code - *refactor/domain/application*  
- [x] **#705** Refactor: Remove idUtils Usage from Domain/Application Code - *refactor/domain/application*
- [x] **#703** Diet Domain: Audit and Remove Legacy Utility Usage - *refactor/domain/code-cleanup*
- [x] **#702** Item-Group Domain: Remove ID Generation and Legacy Utility Usage - *refactor/domain/code-cleanup*
- [x] **#701** Item Domain: Remove ID Generation and Legacy Utility Usage - *refactor/domain/code-cleanup*
- [x] **#700** Meal Domain: Remove ID Generation and Legacy Utility Usage - *refactor/domain/code-cleanup*
- [x] **#654** Issue: Remove ID Generation and Legacy Utility Usage from Domain Code - *refactor/domain/code-cleanup*

#### 🔧 Error Handling & Application Layer
- [x] **#662** Issue: Audit and Refactor Application Error Handling for handleApiError Context - *refactor/application/error-handling*
- [x] **#661** Issue: Add Custom Error Classes for Domain Invariants and Business Rules - *refactor/domain/error-handling*
- [ ] **#663** Issue: Refactor Orchestration Logic to Keep Business Rules in Domain - *refactor/domain/application*

#### 📋 Testing & Documentation
- [ ] **#704** Diet Domain: Update Tests and Documentation After Refactor - *documentation/refactor/domain/test*
- [x] **#660** Issue: Review and Improve Domain Test Coverage - *domain/test*

#### 🎨 UI & Features  
- [ ] **#739** Implement or Remove Toast Settings - *feature/medium complexity*
- [x] **#693** Implement complex group support - *feature*
- [ ] **#620** Enhance Toast System - *feature*
- [ ] **#621** [nitpick] Consider adding focus trapping and return focus to the triggering element when the modal closes to improve accessibility - *feature/low complexity*
- [ ] **#624** Implement priority sorting if needed (avoid infinite loading toasts preventing others) - *refinement/low complexity/ui*
- [ ] **#559** Add item rebalance functionality - *feature*
- [ ] **#558** Add item merge functionality - *feature*
- [x] **#520** Add datepicker to top bar - *improvement/medium complexity*
- [x] **#508** Determine whether to use <Show when/> for modals in general or just remove all Shows - *improvement/low complexity*
- [x] **#470** Handle non-simple groups on handleNewItemGroup - *improvement/medium complexity*
- [x] **#450** make day/TopBar a common component - *improvement/medium complexity*
- [x] **#425** implement setRecipe - *improvement/low complexity*
- [x] **#396** Determine whether to early return from modals in general or just remove all ifs - *improvement/low complexity*
- [x] **#372** Optimize recentFoods fetching: currently, if recentFood is too old (exceeding fetch limit), it will not be fetched and will not be searchable - *improvement/medium complexity*
- [ ] **#366** Improve responsiveness on profile page - *feature/medium complexity*
#### 🚀 Mobile App Development
- [x] **#892** Chart Architecture Optimization with Tabs/Accordion Interface - *feature/complexity-high/performance/ui/needs-design/mobile*
- [ ] **#881** Research: Push notifications for mobile app - *idea/mobile*
- [ ] **#876** Add automated testing for Android builds - *complexity-medium/task/mobile*
- [ ] **#875** Optimize APK size and performance - *improvement/complexity-medium/performance/mobile*
- [ ] **#874** Configure Android release build signing - *complexity-medium/task/mobile*
- [ ] **#873** Configure Android release build signing - *complexity-medium/task/mobile* 
- [ ] **#871** Optimize UI for mobile touch interactions - *improvement/complexity-medium/ui/mobile*
#### 🏗️ Item/ItemGroup Unification Epic
- [x] **#887** Phase 2: Infrastructure Layer for Item/ItemGroup Unification - *complexity-high/refactor/infra/architecture*
- [x] **#885** Phase 4: UI Migration for Item/ItemGroup Unification - *complexity-very-high/refactor/ui*
- [x] **#884** Phase 3: Application Layer for Item/ItemGroup Unification - *complexity-high/refactor/architecture*
- [x] **#883** Phase 1: Schema & Domain Layer for Item/ItemGroup Unification - *feature/complexity-very-high/refactor/architecture*
- [ ] **#882** Epic: Item/ItemGroup Unification for Recursive Recipes - *complexity-very-high/epic/refactor/architecture*
#### 🎨 UI Components & Standardization
- [x] **#852** Unify Copy Button Components - *complexity-low/refactor/ui*
- [x] **#851** Standardize Button Styles with Reusable Components - *improvement/complexity-low/ui*
- [x] **#850** Create Reusable ContextMenuItem Components - *complexity-low/refactor/ui*
#### 🔧 Additional Domain & Schema Work
- [ ] **#600** Use macroNutrientsSchema for other schemas that need macro nutrients - *refactor/domain*
- [ ] **#585** OpenTelemetry (OTel) Integration - *feature/epic*
#### 📋 Planning & Coordination
- [ ] **#865** 📋 v0.13.0 Execution Plan: Critical Domain Refactoring - *documentation/task/epic*

#### 🧪 Investigation & Quality
- [ ] **#708** Explore Replacing Internal Utilities with External Libraries - *improvement/high complexity/needs-investigation*
- [x] **#758** [improvement] Outdated App Version in Build Script - *improvement/low complexity/needs-investigation/build*

#### 📊 Epic & Coordination
- [ ] **#735** Epic: Batch Results Review and Copilot Suggestions - *epic*

---

### 🎯 v0.14.0 - Architecture Improvements and Core Refactoring
**Status**: Planning Phase  
**Focus**: Code quality, error handling, application layer improvements, and technical debt resolution

#### 🏗️ Application Layer & Architecture
- [ ] **#779** Move business logic from UI to application/hooks layer - *improvement/high complexity/refactor/ui/backend*
- [x] **#515** Depency injection for repositories on all application files - *improvement/high complexity*
- [ ] **#468** Make item group not a subcollection of day - *improvement/high complexity*
- [ ] **#467** Make meal not a subcollection of day - *improvement/high complexity*
- [x] **#466** Change upserts to inserts on the entire app - *improvement/medium complexity*
- [x] **#461** Delete old days table and rename days_test to days - *improvement/medium complexity*

#### 🧩 Component Architecture & UI Improvements
- [x] **#622** Unify Modal State Management Across Application - *epic*
- [ ] **#502** Create module for favorites - *improvement/medium complexity*
- [x] **#500** Make meal names editable and persistent by user - *feature/refinement/medium complexity*
- [ ] **#303** Reenable drag and drop - *duplicate/feature/medium complexity*
- [ ] **#302** Enable changing target calories directly (and update macros accordingly) - *improvement/medium complexity*
- [ ] **#250** Add barcode button and handle barcode scan for RecipeEditModal - *duplicate/feature/medium complexity*

#### 🔍 Search & Features
- [ ] **#761** Search Recipes by Ingredients - *feature/question*
- [ ] **#760** Fuzzy Matching for Recipe Search - *feature/question*
- [ ] **#71** Button to reduce and increase target macros by 10% - *feature/medium complexity*

#### 📊 Performance & Build
- [ ] **#769** Remove or gate console.debug statements in production - *low complexity/performance/chore/refactor*
- [ ] **#754** [improvement] Large Bundle/Chunk Size Warnings - *improvement/medium complexity/needs-investigation/performance*
- [ ] **#755** [improvement] Deoptimized Babel Styling for Large Vendor Files - *improvement/low complexity/needs-investigation/build*

#### 🧪 Testing & Quality Assurance
- [ ] **#683** Issue: Expand Audits to Cover Test Structure, Error Propagation, and Integration Points - *test/error-handling*
- [ ] **#670** Issue: Review and Improve Test Coverage for UI Logic and Shared Components - *ui/test*
- [ ] **#666** Issue: Review Application-Level Test Coverage and Integration Points
- [ ] **#665** Issue: Create Application Audit Files for Complex Modules
- [ ] **#664** Issue: Expand Audit to Async Flows, Side Effects, and Error Propagation
- [x] **#627** These tests depend on Solid.js reactivity and queue processing - *medium complexity/infra/test*

#### 🏛️ Domain-Driven Design (DDD) Architecture
- [ ] **#682** Issue: Ensure All Business Rules and Invariants Are Encapsulated in Domain Entities/Value Objects - *domain/architecture*
- [ ] **#678** Issue: Involve Domain Experts for Ongoing Model Refinement - *domain/architecture*
- [ ] **#673** Issue: Clarify Migration Plan and Document Shared Module Boundaries - *documentation/architecture*
- [ ] **#668** Issue: Move Business Logic Out of UI Components - *refactor/ui/architecture*

#### 💡 Ideas & Future Planning
- [ ] **#421** Make current modal present on url so that back button works on mobile - *idea*
- [ ] **#420** Add boolean to recipes "public" so that other users can search them - *idea*
- [x] **#385** Rename to TemplateItemContext - *improvement/low complexity*
#### 🚀 Mobile App Development
- [ ] **#886** Phase 5: Cleanup and Optimization for Item/ItemGroup Unification - *complexity-medium/performance/refactor*
- [ ] **#878** Implement offline data storage with Capacitor Storage - *feature/complexity-high/mobile*
- [ ] **#877** Integrate Capacitor Camera plugin for food scanning - *feature/complexity-high/mobile*
#### 🧩 Advanced Component Architecture
- [ ] **#853** [INVESTIGATION] Create Base Layout Component for View Components - *complexity-medium/refactor/ui*
- [ ] **#849** Create Reusable MacroChart Component for MacroEvolution - *complexity-medium/refactor/ui*
- [x] **#848** Unify Clipboard Copy/Paste Logic Across Edit Views - *complexity-medium/refactor/ui*
#### 🔧 Technical Debt & Refactoring
- [x] **#802** Simplify types in ConfirmModalContext using Accessor types - *complexity-medium/needs-investigation/refactor/ui*
- [x] **#800** Use repository pattern in ItemGroupView - *improvement/complexity-medium/refactor/ui*
- [x] **#799** Use dateUtils in TargetDayPicker for consistent date handling - *improvement/complexity-low/refactor/ui*
- [x] **#797** Replace userDays with userDayIndexes in day-diet infrastructure - *improvement/complexity-medium/refactor/backend*
- [x] **#796** Replace itemEditModalVisible with a derived signal in RecipeEditModal - *improvement/complexity-low/refactor/ui*
- [ ] **#795** Rename property to in recipe item domain - *improvement/complexity-low/refactor/backend*
- [ ] **#794** Rename MacroProfileComp to MacroProfile for consistency - *documentation/complexity-medium/refactor/ui*
- [ ] **#793** Remove nullability from insertDayDiet return type - *improvement/complexity-medium/refactor/backend*
- [ ] **#792** Remove idUtils and migrate all ID generation to the database - *improvement/complexity-high/refactor/backend*
- [ ] **#790** Remove ID generation from createItem and delegate to database - *improvement/complexity-medium/refactor/backend*
- [ ] **#789** Remove all console.error from UI components and move error handling to application layer - *improvement/complexity-medium/refactor/ui*
- [ ] **#788** Refactor conversion from template type to group/item types in TemplateSearchResults - *improvement/complexity-medium/refactor/ui/backend*
- [x] **#787** Propagate actions signal in ConfirmModalContext for improved state management - *improvement/complexity-low/ui*
- [ ] **#781** Create a reusable exhaustive check function in supabaseFoodRepository - *complexity-medium/refactor/backend*
#### 📋 Planning & Coordination
- [ ] **#867** v0.14.0 Execution Plan: Architecture Improvements - *epic*
- [ ] **#866** 🎯 v0.14.0 Execution Plan: Architecture Improvements - *task/epic*

---

### 🎯 v0.15.0 - Major New Features and Advanced Functionality  
**Status**: Future Planning  
**Focus**: Linear Programming diet optimization, advanced UI features, and comprehensive feature development

#### 🚀 Performance & Optimization
- [ ] **#767** Batch fetch foods and recipes in fetchRecentsForModal - *medium complexity/performance/data-consumption/refactor/backend*
- [x] **#765** Add loading indicators for slow fetches in search modals - *improvement/low complexity/performance/refactor/ui*
- [ ] **#757** [improvement] Sourcemap Resolution Errors During Build - *improvement/medium complexity/needs-investigation/build*
- [ ] **#756** [improvement] Consider Upgrading pnpm Version - *improvement/low complexity/needs-investigation/build*

#### 🎯 Advanced Features & User Experience
- [ ] **#734** Add sorting options to template search - *feature/medium complexity*
- [ ] **#695** Allow user to edit recipes inside recipes - *feature/medium complexity*
- [ ] **#557** Make settings persistent on supabase - *feature*
- [ ] **#556** Create simple Settings page with one option - *feature*
- [ ] **#555** User settings - *feature/epic*
- [ ] **#554** Support adding more than one item at a time? - *feature*
- [ ] **#552** Create Settings for MacroOverflow warnings - *feature*
- [ ] **#486** Implement multiple item copy system (select and then copy) - *feature/refinement*
- [ ] **#437** Delete recipes when unlinked from a group and no other group uses it - *improvement/medium complexity/data-consumption*
- [ ] **#423** Prevent user A editing recipe from user B (copy paste) - *feature*
- [ ] **#413** Add portion quantity - *feature*
- [ ] **#411** Create Add new recipe flow - *feature/high complexity*
- [x] **#174** Unify Recipe and Recipe components into a single component? - *improvement/medium complexity/high complexity*
- [ ] **#95** Allow changing percentage directly - *feature*
- [ ] **#28** Make it possible to drag and drop mealitems across meals - *feature/medium complexity*

#### 🐛 Bug Fixes & Issues Investigation
- [ ] **#416** On some devices, dark theme behaves weirdly for fields - *bug/medium complexity/needs-investigation*
#### 🚀 Mobile App Development
- [ ] **#880** Create unified mobile build scripts - *improvement/complexity-medium/mobile*
- [ ] **#879** Add iOS platform support - *feature/complexity-high/mobile*
#### 🤖 Advanced Features & AI
- [ ] **#864** Feature: Automatic Diet Adjustment via Linear Programming (LP Solver) - *feature/complexity-very-high/ui/backend/api*
- [ ] **#843** Implement Intelligent Macro Adjustment Recommendations - *feature/refinement/complexity-very-high/idea/ui/todo :spiral_notepad:*
#### 📊 Performance & Monitoring
- [ ] **#831** Create Migration Documentation and Deployment Strategy - *documentation/refinement/complexity-medium/task/idea/todo :spiral_notepad:*
- [ ] **#830** Add Performance Monitoring for New Features - *refinement/complexity-high/task/idea/performance/backend/todo :spiral_notepad:*
- [ ] **#829** Add Error Context Preservation to Simplified Error Handling - *refinement/complexity-high/idea/refactor/backend/todo :spiral_notepad:*
#### 📋 Planning & Coordination
- [ ] **#868** 🎯 v0.15.0 Execution Plan: Advanced Features & User Experience - *task/epic*

---

### 🎯 v0.16.0 - Future Features and Planned Improvements
**Status**: Long-term Planning  
**Focus**: Advanced functionality, user experience enhancements, and system optimizations

#### 🌐 Internationalization & Accessibility  
- [ ] **#776** Standardize Custom Error Handling in showError Across Codebase - *improvement/medium complexity/needs-investigation/refactor/ui/error-handling*
- [ ] **#686** UI Language (locales, translations) - *feature/epic*
- [ ] **#236** Centralize and Modularize All UI Strings for Future Translation Support - *improvement/medium complexity/high complexity/chore/refactor/ui*
- [ ] **#143** Refactor divs to semantic tags - *improvement/medium complexity*

#### 🎨 Advanced UI & Polish
- [ ] **#294** Create discriminate union type for Male and Female body measures - *improvement*
- [x] **#182** Add support for nested groups and recipes (recursive schema) - *feature/refinement/very high complexity*
- [ ] **#70** Evolution of muscle measures of the body - *feature/refinement*
- [ ] **#43** Implement free day/meal - *feature/refinement*
- [ ] **#42** Create and add a Logo - *feature*
#### 🔧 Technical Improvements & Refactoring
- [ ] **#854** [INVESTIGATION] Create Generic List Component - *complexity-high/refactor*
- [ ] **#836** Add Progressive Macro Target System with Configurable Thresholds - *feature/refinement/complexity-medium/idea/ui/todo :spiral_notepad:*
- [ ] **#835** Implement Enhanced Backend Outage State Management with Auto-Recovery - *feature/refinement/complexity-medium/idea/backend/todo :spiral_notepad:*
- [ ] **#834** Add Dismissible Backend Outage Banner with Enhanced UX - *feature/refinement/complexity-medium/idea/ui/accessibility/todo :spiral_notepad:*
- [ ] **#832** Implement Portuguese Diacritic-Insensitive Search Integration - *feature/refinement/complexity-medium/idea/ui/todo :spiral_notepad:*
- [ ] **#811** Weight/Measure: preserve or allow changing time when editing date - *feature*
- [ ] **#801** Update documentation and tests for all changes - *documentation/complexity-medium/chore*
- [ ] **#798** Set autofocus based on device type in TemplateSearchModal - *improvement/complexity-low/refactor/ui*
- [ ] **#791** Remove ID generation from createRecipeItem and delegate to database - *improvement/complexity-medium/refactor/backend*
- [ ] **#786** Move to server onSave(newProfile) in UserInfoCapsule - *improvement/complexity-high/refactor/backend/security*
- [ ] **#785** Move function in MealEditView to a more appropriate module - *improvement/complexity-low/refactor/ui*
- [ ] **#784** Monitor Zod discriminated union deprecation and prepare for schema refactor - *improvement/complexity-low/backend*
- [ ] **#782** Create a reusable exhaustive check function in toast UI module - *improvement/complexity-medium/refactor/ui*
- [ ] **#780** Change target_day to Supabase date type for improved compatibility - *improvement/complexity-low/backend*
- [x] **#606** Compare macros when they are implemented in the recipe - *refinement/domain*
#### 🗂️ Legacy & Cleanup
- [x] **#860** : default handlers for ItemView - *todo :spiral_notepad:*
- [ ] **#859** Remove deprecated props and their usages - *todo :spiral_notepad:*
#### 📋 Planning & Coordination
- [ ] **#869** 🎯 v0.16.0 Execution Plan: Polish & Future Foundation - *task/epic*

---

### 🗂️ DDD Milestone - Domain-Driven Design Foundation
**Status**: Ongoing Architecture Initiative  
**Focus**: Establishing proper domain boundaries and clean architecture

#### 🏛️ Domain Architecture & Boundaries
- [ ] **#447** Domain Driven Design (DDD) - *very high complexity/epic*
- [ ] **#684** Issue: Document and Communicate Architectural Decisions and DDD Boundaries for New Contributors - *documentation/architecture*
- [x] **#681** Issue: Make Bounded Contexts Explicit and Avoid Generic Technical Abstractions - *domain/architecture*
- [ ] **#680** Issue: Use Value Objects Instead of Entities Where Identity Is Not Essential - *domain/architecture*
- [ ] **#679** Issue: Avoid Overengineering (CQRS, Event Sourcing, Hexagonal) Unless Justified - *domain/architecture*
- [ ] **#677** Issue: Review Cross-Module Dependencies and Domain Events - *domain/architecture*
- [ ] **#676** Issue: Review Shared Code Test Coverage and DDD Violations
- [ ] **#675** Issue: Map All Shared/Legacy Code Usage and Document Migration Blockers - *documentation/code-cleanup*
- [ ] **#674** Issue: Propose and Execute Migration Steps for Remaining Legacy Utilities - *refactor/code-cleanup*

#### 🔧 Domain & Application Layer Improvements
- [ ] **#671** Issue: Unify or Consolidate Edit Modals/Views in UI Sections - *refactor/ui*
- [ ] **#669** Issue: Audit UI Sections for Business Logic Leakage and Duplication
- [ ] **#667** Issue: Refactor Duplicated Logic in UI/Sections into Shared Hooks/Utilities - *refactor/ui/code-cleanup*
- [x] **#659** Issue: Introduce or Strengthen Domain Layers in Weak Modules - *refactor/domain*
- [ ] **#658** Issue: Clarify and Document Bounded Contexts; Split Large Modules if Needed - *documentation/domain/architecture*
- [ ] **#657** Issue: Expand Use of Value Objects for Identity-less Concepts - *refactor/domain*
- [ ] **#656** Issue: Standardize and Document Repository Interfaces and Error Types - *documentation/domain/error-handling*
- [ ] **#655** Issue: Remove ID Generation and Legacy Utility Usage from Domain Code - *epic*
- [ ] **#459** Move utils/* to DDD - *improvement*

---

---
### 🗃️ Backlog (Prioritized) - High Priority Technical Debt
**Status**: Important improvements with clear implementation path  
**Focus**: Code quality, performance, and development experience improvements
#### 🔧 Code Quality & Refactoring
- [x] **#824** Consider property-based testing for macro overflow logic if logic becomes more complex - *todo :spiral_notepad:*
- [x] **#820** Enforce removal of unused exports with ts-unused-exports - *improvement/complexity-low/build*
- [x] **#819** Use CopyButton in all buttons with CopyIcon - *complexity-medium/refactor/ui*
- [x] **#818** Refactor: Replace all console.debug with createDebug - *complexity-medium/refactor*
- [ ] **#816** Refactor currentDayDiet to SolidJS Resource and add Show fallback in diet.tsx - *complexity-medium/refactor/ui*
---
### 🚨 Unassigned Issues - Needs Milestone Assignment
**Status**: Critical issues requiring immediate milestone assignment  
**Focus**: Performance and self-improvement
#### 🔥 Critical Performance
- [x] **#888** Performance: Profile page extremely slow on mobile devices - *complexity-high/performance/epic/ui/mobile*
#### 📋 Meta Issues  
- [ ] **#488** Centralized Roadmap - *improvement/epic* (self-reference)
---
### 🗃️ Backlog - Future Considerations
**Status**: Ideas and long-term features  
**Focus**: Advanced features requiring significant development effort

#### 🔮 Advanced Features (Very High Complexity)
- [ ] **#434** Add undo function - *very high complexity/idea*
- [ ] **#285** Allow registering evolution of muscle sizes - *duplicate/feature/very high complexity*
- [ ] **#283** Calculate IMC - *feature/very high complexity*
- [ ] **#282** Make predicitons how close in months/days user is to get to the weight target based on past averages - *feature/very high complexity*
- [ ] **#272** Differentiate types of fat (monosaturated and monounsaturated) - *feature/very high complexity*
- [ ] **#132** Import/Export Diet - *feature/very high complexity*
- [ ] **#82** Add custom food manually (only with macros) - *feature/very high complexity*
- [ ] **#81** Edit food on-the-fly in case food data retrieved by the api is wrong - *feature/very high complexity*
- [ ] **#41** Weekly macros (support for carbo cycles and others) - *feature/refinement/very high complexity*

#### 💡 Ideas for Future Exploration
- [ ] **#438** Sync recipe name into linked item group (not sure when tho) - *idea*
- [ ] **#842** Enhance Diacritics Removal Test Coverage - *refinement/complexity-low/task/idea/chore/todo :spiral_notepad:*
- [x] **#841** Fix Indentation and Add Input Validation to Cat Scripts - *refinement/complexity-low/task/idea/chore/todo :spiral_notepad:*
- [ ] **#840** Restore and Fix semver.sh Test Coverage - *refinement/complexity-medium/task/idea/chore/todo :spiral_notepad:*
- [x] **#839** Enhance semver.sh with Robust Error Handling and Retry Logic - *refinement/complexity-medium/task/idea/chore/todo :spiral_notepad:*
- [ ] **#838** Optimize Macro Overflow Calculations for Performance - *refinement/complexity-medium/task/idea/performance/todo :spiral_notepad:*
- [ ] **#837** Optimize Recent Items Search Performance - *refinement/complexity-medium/task/idea/performance/todo :spiral_notepad:*
- [ ] **#833** Add Comprehensive Integration Tests for Search Logic Refactor - *refinement/complexity-high/task/idea/todo :spiral_notepad:*

---

#### 🔥 Critical Performance Issues
- **#767** Batch fetch foods and recipes in fetchRecentsForModal - *v0.15.0*
- **#765** Add loading indicators for slow fetches - *v0.15.0*  
- **#769** Remove console.debug statements in production - *v0.14.0*
- **#754** Large Bundle/Chunk Size Warnings - *v0.14.0*

#### 📊 Build & Infrastructure Performance
- **#757** Sourcemap Resolution Errors - *v0.15.0*
- **#756** Consider Upgrading pnpm Version - *v0.15.0*
- **#755** Deoptimized Babel Styling Warnings - *v0.14.0*
- **#758** Outdated App Version in Build Script - *v0.13.0*

#### 🎯 User Experience Improvements
- **#366** Improve responsiveness on profile page - *v0.13.0*
- **#416** Dark theme field issues on some devices - *v0.15.0*
- **#421** Modal URL integration for back button - *v0.14.0* (*idea*)

#### 🔧 Major Refactoring Initiatives
- **#730** Reduce schema duplication across domain - *v0.13.0*
- **#727** Unify macro/weight calculation logic - *v0.13.0*
- **#733** Recipe-based macro calculations - *v0.13.0*
- **#779** Move business logic from UI to application layer - *v0.14.0*
- **#515** Dependency injection for repositories - *v0.14.0*

#### 🧹 Legacy Code Cleanup
- **#672** Remove Legacy Utility Usage (Meta-issue) - *v0.13.0*
- **#707** Remove deepCopy Usage - *v0.13.0*
- **#706** Remove macroMath Usage - *v0.13.0*
- **#705** Remove idUtils Usage - *v0.13.0*
- **#703** Diet Domain Legacy Cleanup - *v0.13.0*
- **#702** Item-Group Domain Legacy Cleanup - *v0.13.0*
- **#701** Item Domain Legacy Cleanup - *v0.13.0*
- **#700** Meal Domain Legacy Cleanup - *v0.13.0*

#### 🏛️ Domain-Driven Design
- **#447** Domain Driven Design (DDD) - *DDD Milestone*
- **#684** Document Architectural Decisions - *DDD Milestone*
- **#682** Encapsulate Business Rules in Domain - *v0.14.0*
- **#681** Make Bounded Contexts Explicit - *DDD Milestone*
- **#680** Use Value Objects Where Appropriate - *DDD Milestone*

#### 🎯 Search Improvements
- **#761** Search Recipes by Ingredients - *v0.14.0*
- **#760** Fuzzy Matching for Recipe Search - *v0.14.0*
- **#743** Recipe search user quantity handling - *v0.12.0*
- **#372** Optimize recentFoods fetching - *v0.13.0*
- **#734** Add sorting options to template search - *v0.15.0*

### 🤖 AI & Automation
*Currently no specific AI/automation issues are open*

#### 📈 Data Layer Improvements  
- **#723** Remove dayId from unnecessary functions - *v0.13.0*
- **#713** Implement proper infrastructure for recent food - *v0.13.0*
- **#468** Make item group not a subcollection of day - *v0.14.0*
- **#467** Make meal not a subcollection of day - *v0.14.0*
- **#466** Change upserts to inserts - *v0.14.0*
- **#461** Delete old days table and rename days_test - *v0.14.0*

#### 🗃️ Settings & User Data
- **#557** Make settings persistent on supabase - *v0.15.0*
- **#556** Create simple Settings page - *v0.15.0*
- **#555** User settings (Epic) - *v0.15.0*
- **#552** Create Settings for MacroOverflow warnings - *v0.15.0*

#### 🖼️ Interface Components & Polish
- **#622** Unify Modal State Management - *v0.14.0* (*epic*)
- **#739** Implement or Remove Toast Settings - *v0.13.0*
- **#620** Enhance Toast System - *v0.13.0*
- **#500** Make meal names editable and persistent - *v0.14.0*
- **#520** Add datepicker to top bar - *v0.13.0*
- **#450** Make day/TopBar a common component - *v0.13.0*

#### 🎭 Accessibility & User Experience
- **#621** Focus trapping and return focus for modals - *v0.13.0*
- **#624** Priority sorting for loading toasts - *v0.13.0*
- **#508** Determine Show when usage for modals - *v0.13.0*
- **#396** Determine early return from modals - *v0.13.0*

#### 🎨 Advanced UI Features
- **#303** Reenable drag and drop - *v0.14.0* (*duplicate*)
- **#302** Enable changing target calories directly - *v0.14.0*
- **#250** Add barcode button to RecipeEditModal - *v0.14.0* (*duplicate*)
- **#71** Button to adjust target macros by 10% - *v0.14.0*

#### 🌐 Internationalization & Accessibility
- **#686** UI Language (locales, translations) - *v0.16.0* (*epic*)
- **#236** Centralize UI Strings for Translation - *v0.16.0*
- **#143** Refactor divs to semantic tags - *v0.16.0*

#### 🛠️ Developer Tools & Quality
- **#708** Explore Replacing Internal Utilities - *v0.13.0*
- **#704** Update Tests and Documentation After Refactor - *v0.13.0*
- **#683** Expand Audits for Test Structure - *v0.14.0*
- **#670** Review Test Coverage for UI Logic - *v0.14.0*
- **#666** Review Application-Level Test Coverage - *v0.14.0*
- **#665** Create Application Audit Files - *v0.14.0*
- **#664** Expand Audit to Async Flows - *v0.14.0*
- **#660** Review Domain Test Coverage - *v0.13.0*
- **#627** Solid.js reactivity test dependencies - *v0.14.0*

#### 📚 Documentation & Architecture
- **#735** Epic: Batch Results Review and Copilot Suggestions - *v0.13.0* (*epic*)
- **#673** Clarify Migration Plan and Document Boundaries - *v0.14.0*
- **#675** Map Shared/Legacy Code Usage - *DDD Milestone*
- **#674** Propose Migration Steps for Legacy Utilities - *DDD Milestone*

#### 🐛 Active Bugs
- **#743** Recipe search preparedMultiplier not respected - *v0.12.0*
- **#416** Dark theme field behavior on some devices - *v0.15.0*

#### 🔧 Error Handling & Quality
- **#776** Standardize Custom Error Handling - *v0.16.0*
- **#662** Audit Application Error Handling - *v0.13.0*
- **#661** Add Custom Error Classes for Domain - *v0.13.0*
- **#663** Refactor Orchestration Logic - *v0.13.0*

#### 🍽️ Recipe & Meal Management
- **#695** Allow editing recipes inside recipes - *v0.15.0*
- **#693** Implement complex group support - *v0.13.0*
- **#559** Add item rebalance functionality - *v0.13.0*
- **#558** Add item merge functionality - *v0.13.0*
- **#486** Implement multiple item copy system - *v0.15.0*
- **#437** Delete recipes when unlinked - *v0.15.0*
- **#425** implement setRecipe - *v0.13.0*
- **#423** Prevent user A editing recipe from user B - *v0.15.0*
- **#413** Add portion quantity - *v0.15.0*
- **#411** Create Add new recipe flow - *v0.15.0*

#### 🎨 Advanced Features
- **#554** Support adding more than one item at a time - *v0.15.0*
- **#502** Create module for favorites - *v0.14.0*
- **#470** Handle non-simple groups - *v0.13.0*
- **#385** Rename to TemplateItemContext - *v0.14.0*
- **#174** Unify Recipe components - *v0.15.0*
- **#95** Allow changing percentage directly - *v0.15.0*
- **#28** Drag and drop mealitems across meals - *v0.15.0*

#### 🌟 Advanced Body & Health Features
- **#294** Discriminate union for Male/Female measures - *v0.16.0*
- **#182** Support for nested groups and recipes - *v0.16.0*
- **#70** Evolution of muscle measures - *v0.16.0*
- **#43** Implement free day/meal - *v0.16.0*
- **#42** Create and add a Logo - *v0.16.0*

#### 🔮 Future Advanced Features
- **#434** Add undo function - *Backlog*
- **#285** Register evolution of muscle sizes - *Backlog*
- **#283** Calculate IMC - *Backlog*
- **#282** Weight target predictions - *Backlog*
- **#272** Differentiate types of fat - *Backlog*
- **#132** Import/Export Diet - *Backlog*
- **#82** Add custom food manually - *Backlog*
- **#81** Edit food on-the-fly - *Backlog*
- **#41** Weekly macros (carbo cycles) - *Backlog*

#### 💡 Ideas for Future Exploration
- **#438** Sync recipe name into linked item group - *v0.13.0* (*idea*)
- **#420** Add boolean to recipes "public" - *v0.14.0* (*idea*)

---

### 🔥 High Priority (v0.12.0-v0.13.0)
1. **Critical bugs** - Issues that prevent normal usage
2. **Domain/application architecture** - Foundation for future features  
3. **Legacy code cleanup** - Maintainability and performance
4. **Essential features** - Core product functionality

### 🎯 Medium Priority (v0.14.0-v0.15.0)  
1. **Component refactoring** - Development efficiency
2. **Advanced features** - Competitive differentiation
3. **Performance optimizations** - User experience
4. **Testing & quality assurance** - Product stability

### ⭐ Low Priority (v0.16.0+)
1. **UX polish** - Premium experience
2. **Internationalization** - Market expansion
3. **Experimental features** - Innovation and differentiation
4. **Future preparation** - Foundation for growth

---

### 🏗️ Architecture Quality
- [ ] Zero schema duplication across domains
- [ ] 95% reduction in legacy utility usage
- [ ] Unified macro/weight calculation system
- [ ] Clean domain/application layer separation
- [ ] Comprehensive error handling with context

### 🔍 Feature Completeness
- [ ] Advanced search functionality operational
- [ ] Complex group support implemented
- [ ] Recipe management fully featured
- [ ] Settings system functional and persistent
- [ ] UI component system standardized

### 🧪 Code Quality & Testing
- [ ] 80%+ test coverage for domain logic
- [ ] 70%+ test coverage for application layer
- [ ] Zero critical bugs in production
- [ ] All DDD violations resolved
- [ ] Complete architectural documentation

### 🎨 User Experience
- [ ] 100% accessibility compliance
- [ ] Responsive design across all devices
- [ ] Intuitive recipe and meal management
- [ ] Fast and reliable search functionality
- [ ] Consistent UI patterns and interactions

---

## 🔄 Update Process

This roadmap will be updated:
- **Weekly** during active milestone development
- **Bi-weekly** for priority adjustments within milestones
- **Monthly** for general review and feature rebalancing
- **Ad-hoc** for critical changes, new bugs, or blockers

**Methodology**: Issues are organized by milestone according to criticality, technical dependencies, and business value. Features are grouped by category to facilitate parallel development and architecture review.

**Last updated**: January 17, 2025  
**Next review**: January 31, 2025
**Total issues included**: 210 open issues

---

### 🔧 Technical Dependencies
- **v0.13.0** is prerequisite for v0.14.0 (architectural foundation)
- **DDD Milestone** issues can be developed in parallel with other milestones
- **Legacy utility removal** must be completed before advanced features
- **Domain refactoring** is critical for performance and maintainability

### 🎯 Development Strategy
- **Parallel tracks**: UI improvements can be developed in parallel with domain refactoring
- **Incremental approach**: Complex features are broken into sub-issues when possible
- **Quality gates**: Testing requirements increase with milestone maturity
- **Documentation driven**: All architectural changes require documentation updates
